planter/planter.sh

```diff
@@ -39,19 +39,13 @@ IMAGE="${IMAGE_NAME}:${TAG}"
 # - WORKSPACE is assumed to be in your current git repo, or alternatively $PWD
 REPO=$(git rev-parse --show-toplevel 2>/dev/null || true)
 REPO=${REPO:-${PWD}}
-# - the bazel cache is in $HOME/.cache/bazel
-#
-# - we could just mount $HOME/.cache/bazel, but unfortunately some rules
-# are still not terribly hermetic and actually need $HOME (rules_node)
-#
-# - $HOME/.docker shouldn't be mounted so we clobber it with a tmpfs.
-# You can have a $HOME/.docker/config.json for logging in to docker hub,
-# which on mac will actually store credentials in your keychain which
-# the container cannot access, this breaks rules_docker. (issues/5736)
+# To build kubernetes, first you checkout github.com/kubernetes/kubernetes
+# to $GOPATH/src/k8s.io/kubernetes and github.com/kubernetes/test-infra
+# to $GOPATH/src/k8s.io/test-infra.
 #
 # - /tmp also needs to be a suitable tmpfs mounted with exec so that bazel
 # can use it when executing various things
-VOLUMES="-v ${REPO}:${REPO} -v ${HOME}:${HOME} --mount type=tmpfs,destination=${HOME}/.docker --tmpfs /tmp:exec,mode=777"
+VOLUMES="-v ${REPO}:${REPO} -v ${HOME}/.cache/bazel:/${HOME}/.cache/bazel -v ${HOME}/.npm:/${HOME}/.npm --tmpfs /tmp:exec,mode=777"
```

Comment:
I removed the docker override, since we won't need it anymore mounting only bazel and npm caches

Reply:
does this do the right thing if `${HOME}/.cache/bazel` or `${HOME}/.npm` don't exist yet?

Reply:
Answer: not exactly. Docker will create these directories on the host if needed, but they'll be owned by root, and so they won't be writable by the user running planter.

Maybe we first create these directories if they don't exist? The alternate is to `chown` them, but I don't like that as much.

Reply:
We should definitely prefer `mkdir` to `chown`

Reply:
@ixdy how about merging this and then doing mkdir in a follow-up?

Reply:
I'm a little hesitant, since this can yield a poor user experience if these directories don't exist yet.